File: solutions/__119_max_profit.go
package main

// 如果在遍历的过程中 有个更小的点 我们记录下来
// 因为如果后面有更大的点 则与这个点相减会比之前的差值更大
func maxProfit(prices []int) int {
	min := prices[0]
	rtv := 0
	n := len(prices)
	for i := 1; i < n; i++ {
		if prices[i]-min > rtv {
			rtv = prices[i] - min // 保存最大差值
		}
		if prices[i] < min {
			min = prices[i]
		}
	}
	return rtv
}

File: solutions/__119_max_profit.cpp
/*
 *      Author: peanutzhen
 *      Created time: 2021/8/22 13:50:21
 */

#include <bits/stdc++.h>

using namespace std;

int maxProfit(vector<int> &prices) {
    int n = (int) prices.size();
    int min_price = INT_MAX;
    int ans = 0;
    for (int i = 0; i < n; ++i) {
        if (prices[i] < min_price)
            min_price = prices[i];
        int tmp = prices[i] - min_price;
        if (tmp > ans)
            ans = tmp;
    }
    return ans;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);


    return 0;
}

